pkg/eval/builtin_fn_misc.go

```diff
@@ -160,7 +161,30 @@ import (
 // bar
 // ```
 
-// TODO(xiaq): Document esleep.
+//elvdoc:fn sleep
+//
+// ```elvish
+// sleep $duration
+// ```
+//
+// Sleep for the specified duration.  A duration string is a possibly signed
+// sequence of decimal numbers, each with optional fraction and a unit suffix,
+// such as "300ms", "-1.5h" or "2h45m". Valid time units are "ns", "us" (or
+// "µs"), "ms", "s", "m", "h". The default time unit is "s" (seconds). Note
```

Comment:
> The default time unit is "s" (seconds)

I'd prefer this to be documented more distinctly, since we only support a single number when there is no unit; `1m2` is forbidden intentionally, for example. Something like:

> Alternatively, the duration can be given as a single number, in which case it is interpreted as a number of seconds.

Reply:
Agreed, but I think a different approach is preferable. Especially since I noticed, as a consequence of thinking about your comment, that the current implementation does not support `float64` values. These two examples should be identical but the second one raises a "wrong type: need string, got number" exception:

```
> sleep 1
> sleep (float64 1)
```